Mathlib/AlgebraicGeometry/Morphisms/LocalClosure.lean

```diff
@@ -0,0 +1,149 @@
+/-
+Copyright (c) 2025 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+
+import Mathlib.AlgebraicGeometry.Morphisms.Basic
+
+/-!
+# Local closure of morphism properties
+
+We define the source local closure of a property `P` wrt. a morphism property `W` and show it
+inherits stability properties from `P`.
+-/
+
+universe u
+
+open CategoryTheory Limits MorphismProperty
+
+namespace AlgebraicGeometry
+
+variable (W : MorphismProperty Scheme.{u})
+
+/-- The source (Zariski-)local closure of `P` is satisfied if there exists
+an open cover of the source on which `P` is satisfied. -/
+def SourceLocalClosure (P : MorphismProperty Scheme.{u}) : MorphismProperty Scheme.{u} :=
+  fun X _ f ↦ ∃ (𝒰 : Scheme.Cover.{u} W X), ∀ (i : 𝒰.J), P (𝒰.map i ≫ f)
+
+namespace SourceLocalClosure
+
+variable (P Q : MorphismProperty Scheme.{u}) {X Y : Scheme.{u}}
+
+/-- A choice of open cover on which `P` is satisfied if `f` satisfies the source local closure
+of `P`. -/
+noncomputable def openCover {f : X ⟶ Y} (hf : SourceLocalClosure W P f) : Scheme.Cover.{u} W X :=
```

Comment:
For these definitions, we could use `hf` as the only explicit variable, so that dot notation would apply.